src/java.base/share/classes/java/text/ListFormat.java

```diff
@@ -0,0 +1,557 @@
+     * placeholders:
+     * <blockquote><pre>
+     * (start_before){0}start_between{1}middle_between{2} ... middle_between{m}end_between{n}(end_after)
+     * </pre></blockquote>
+     * If the number of elements in the input string list is either 2 or 3, and the
+     * pattern for those (either "two" or "three", respectively) is not empty, the
+     * pattern is used as it is.
+     * Following table shows patterns array which is equivalent to
+     * {@code STANDARD} type, {@code FULL} style in US English:
+     * <table class="striped">
+     * <caption style="display:none">Standard/Full Patterns in US English</caption>
+     * <thead>
+     * <tr><th scope="col">Pattern Kind</th>
+     *     <th scope="col">Pattern String</th></tr>
+     * </thead>
+     * <tbody>
+     * <tr><th scope="row" style="text-align:left">start</th>
+     *     <td>"{0}, {1}"</td>
+     * <tr><th scope="row" style="text-align:left">middle</th>
+     *     <td>"{0}, {1}"</td>
+     * <tr><th scope="row" style="text-align:left">end</th>
+     *     <td>"{0}, and {1}"</td>
+     * <tr><th scope="row" style="text-align:left">2</th>
+     *     <td>"{0} and {1}"</td>
+     * <tr><th scope="row" style="text-align:left">3</th>
+     *     <td>""</td>
+     * </tbody>
+     * </table>
+     * Here are the resulting formatted strings with the above pattern array.
+     * <table class="striped">
+     * <caption style="display:none">Formatting examples</caption>
+     * <thead>
+     * <tr><th scope="col">Input String Array</th>
+     *     <th scope="col">Formatted String</th></tr>
+     * </thead>
+     * <tbody>
+     * <tr><th scope="row" style="text-align:left">["Foo", "Bar", "Baz", "Qux"]</th>
+     *     <td>"Foo, Bar, Baz, and Qux"</td>
+     * <tr><th scope="row" style="text-align:left">["Foo", "Bar", "Baz"]</th>
+     *     <td>"Foo, Bar, and Baz"</td>
+     * <tr><th scope="row" style="text-align:left">["Foo", "Bar"]</th>
+     *     <td>"Foo and Bar"</td>
+     * <tr><th scope="row" style="text-align:left">["Foo"]</th>
+     *     <td>"Foo"</td>
+     * </tbody>
+     * </table>
+     *
+     * @param patterns array of patterns, not null
+     * @throws IllegalArgumentException if the length {@code patterns} array is less than 5, or
+     *          any of {@code start}, {@code middle}, {@code end} patterns cannot be parsed.
+     * @throws NullPointerException if {@code patterns} is null.
+     */
+    public static ListFormat getInstance(String[] patterns) {
+        Objects.requireNonNull(patterns);
+        if (patterns.length != PATTERN_ARRAY_LENGTH) {
+            throw new IllegalArgumentException("Pattern array length should be " + PATTERN_ARRAY_LENGTH);
+        }
+        return new ListFormat(Locale.ROOT, Arrays.copyOf(patterns, PATTERN_ARRAY_LENGTH));
+    }
+
+    /**
+     * {@return the string that consists of the input strings, concatenated with the
+     * patterns specified, or derived from {@code Locale}, {@code Type}, and {@code Style}}
```

Comment:
```suggestion
     * patterns of this {@code ListFormat}
```
Perhaps link to the ListFormat method that returns the List/array of the 5 patterns.